blocks/library/cover-image/index.js

```diff
@@ -196,23 +149,25 @@ export const settings = {
 				style={ style }
 				className={ classes }
 			>
-				{ title || isSelected ? (
-					<RichText
-						tagName="p"
-						className="wp-block-cover-image-text"
-						placeholder={ __( 'Write title…' ) }
-						value={ title }
-						onChange={ ( value ) => setAttributes( { title: value } ) }
-						isSelected={ isSelected }
-						inlineToolbar
+				<div className="wp-block-cover-image__inner-container">
+					<InnerBlocks
+						template={ [
+							[ 'core/paragraph', {
+								align: 'center',
+								fontSize: 37,
+								placeholder: 'Write title…',
+								textColor: '#fff',
+							} ],
+						] }
+						allowedBlockNames={ [ 'core/button', 'core/heading', 'core/paragraph', 'core/subhead' ] }
```

Comment:
Should we move it to the settings object in `registerBlockType` to make it extensible? :)

Reply:
I meant both `template` and `allowedBlockNames`.

Reply:
As noted in the description, this allows it to be customizable using toolbar/inspector controls.

But this makes me think of something. We should consider the `template` prop here as a `defaultTemplate` because we already have support for nested template. by setting a CPT template with a nested template for the cover image for instance. I think we should do something here where the template prop is used (override it by the global template if available or something)

Reply:
I’m missing something here, how can I override those values for Cover Image block? :smile:
I couldn't digest this from this PR.

Reply:
Can we switch this API to be `allowedBlocks`?

Reply:
Hi @youknowriad,

> But this makes me think of something. We should consider the template prop here as a defaultTemplate because we already have support for nested template. by setting a CPT template with a nested template for the cover image for instance. I think we should do something here where the template prop is used (override it by the global template if available or something).

When we have a CPT template and we set content for a nested block, ( e.g: we set a cover image with a header and a paragraph), during the post-load the cover image is created and the blocks are added inside it. So when our inner logic executes the cover image block already has content, so the template will not be used. It will be the same as when we load a post with a cover image that already has blocks inside it. The template only applies to empty blocks.
So both approaches can work together without problems if a global template exists it already overwrites this setting :)